plugins/modules/ndo_service_device_cluster.py

```diff
@@ -0,0 +1,641 @@
+
+    mso.exit_json()
+
+
+def get_interfaces_payload(mso, mso_template, interfaces, references, mso_schema_cache, mso_template_cache):
+    def get_object_uuid_from_schema(interface_type, uuid, interface):
+        if uuid:
+            return uuid
+
+        name = interface.get("name")
+        template = interface.get("template")
+        template_id = interface.get("template_id")
+        schema_name = interface.get("schema")
+        schema_id = interface.get("schema_id")
+
+        get_schema = mso_schema_cache.get_schema(mso, schema_name, schema_id, template, template_id)
+
+        if interface_type == "bd":
+            get_schema.set_template_bd(name, fail_module=True)
+            return get_schema.schema_objects.get("template_bd").details.get("uuid")
+        elif interface_type == "l3out":
+            get_schema.set_template_external_epg(name, fail_module=True)
+            return get_schema.schema_objects.get("template_external_epg").details.get("uuid")
+
+    def get_object_uuid_from_template(object_type, uuid, obj):
+        if uuid:
+            return uuid
+
+        is_empty = check_if_all_elements_are_none(obj.values()) if obj else True
+        if not is_empty:
+            name = obj.get("name")
+            template = obj.get("template")
+            template_id = obj.get("template_id")
+
+            get_template = mso_template_cache.get_template(mso, "tenant", template, template_id)
+
+            if object_type == "ipsla":
+                return mso_template.get_tenant_policy_uuid(get_template, name, "ipslaMonitoringPolicies")
+            else:
+                return mso_template.get_tenant_policy_uuid(get_template, name, "qosPolicies")
+
+    payload = []
+    for interface in interfaces:
+        mso_template.check_template_when_name_is_provided(interface.get("ipsla_monitoring_policy"))
+        mso_template.check_template_when_name_is_provided(interface.get("qos_policy"))
+        interface_payload = {"name": interface.get("name")}
+        interface_payload["deviceInterfaceType"] = interface.get("type")
+        interface_payload["thresholdForRedirectDestination"] = interface.get("threshold_for_redirect_destination")
+        interface_payload[references.get(interface.get("type"))] = get_object_uuid_from_schema(
+            interface.get("type"), interface.get("interface_uuid"), interface.get("interface")
+        )
+        interface_payload["redirect"] = interface.get("redirect")
+        interface_payload["redirect"] = True
+        interface_payload["isAdvancedIntfConfig"] = True
+        interface_payload["ipslaMonitoringRef"] = get_object_uuid_from_template(
+            "ipsla", interface.get("ipsla_monitoring_policy_uuid"), interface.get("ipsla_monitoring_policy")
+        )
+        interface_payload["advancedIntfConfig"] = {
+            "thresholdForRedirectDestination": True,
+            "rewriteSourceMac": interface.get("rewrite_source_mac"),
```

Comment:
are there any issues when certain configuration items are not provided and send as None? is the mso.sanitize function already cleaning those None value or are they accepted? In the test files I see for creation we test all the time with all non required attributes set. 

```
preferred_group: true
            rewrite_source_mac: false
            config_static_mac: false
            is_backup_redirect_ip: true
            load_balance_hashing: sourceIP
            pod_aware_redirection: false
            resilient_hashing: true
            tag_based_sorting: false
            min_threshold: 10
            max_threshold: 100
            threshold_down_action: permit
```

should we add test case that creates an object with only those required attributes?

Reply:
ok